### PR TITLE
The edge filter example contains all implemented filters apart from t…

### PR DESCRIPTION
**Pull request by tv3141**
…he Prewitt filter. Added the Prewitt filter to the example and compared it to the Scharr filter showing that Prewitt is even less rotationally invarient than the Sobel filter. Also added description.

### CODE REVIEW

File: doc/examples/plot_edge_filter.py [hunk 1]
```diff
@@ -36,17 +36,21 @@
...
-direction of the gradient is close to diagonal, and for regions with high
-spatial frequencies.
+variance than the Sobel filter that is in turn better than the Prewitt filter
+[1]_ [2]_ [3]_. The difference between the Prewitt and Sobel filters and the
+Scharr filter is illustrated below on an image that is the discretization of a
+rotation-invariant continuous function. The discrepancy between the Prewitt and
+Sobel filters, and the Scharr filter is stronger for regions of the image where
+the direction of the gradient is close to diagonal, and for regions with high
+spatial frequencies. The Sobel filter is less rotationally invariant than the
+Sobel filter.
```
Comment (emmanuelle) [inline doc/examples/plot_edge_filter.py]: is there a typo here? Did you mean Scharr?

File: doc/examples/plot_edge_filter.py [hunk 2]
```diff
@@ -55,25 +59,38 @@
...
+ax3.imshow(img, cmap=plt.cm.gray)
+ax3.set_title('Original image')
 ax3.axis('off')
 
+diff_scharr_prewitt = edge_scharr - edge_prewitt
+diff_scharr_sobel = edge_scharr - edge_sobel
+max_diff = np.max(np.maximum(diff_scharr_prewitt, diff_scharr_sobel))
+
+ax4.imshow(diff_scharr_prewitt, cmap=plt.cm.jet, vmax=max_diff)
+ax4.set_title('difference (Scharr - Prewitt)')
```
Comment (emmanuelle) [inline doc/examples/plot_edge_filter.py]: titles are a bit too large, "Scharr - Prewitt" should be clear enough I think.

**Comment by emmanuelle** [comment]
Thank you very much for this addition! I just made couple of comments. I wonder who is behind the cryptic name of @tv3141 :-)

**Comment by emmanuelle** [comment]
Since the output of the 3 edge filters look exactly the same, a possibility would be to plot only one of them (like Scharr), the original image, and then the differences between Scharr and Sobel, and Scharr and Prewitt. Then we'd still have a 2x2 image, which is easier and faster to look at. I don't have strong opinions about this point, just a suggestion.

**Comment by jni** [comment]
@emmanuelle I like that suggestion! @tv3141 do you think you can tackle this?

**Comment by emmanuelle** [comment]
Looks good to me, thank you very much for the changes. Travis-CI is stalling for the Python 2.6 build, I've restarted it and then I'll merge.

**Comment by blink1073** [comment]
@emmanuelle, the 2.6 problem is known: #1692.

**Comment by blink1073** [comment]
(Not sure what to do about it though).

**Comment by JDWarner** [comment]
@tv3141, @blink1073 has now fixed the Travis issue on master. A quick rebase will be needed for Travis to pick up the changes (or @emmanuelle can just approve this as-is).

**Comment by ahojnnes** [comment]
I am so bold to merge this. This looks great. Thanks!

**Pull request merged**